src/pybind/mgr/cephadm/agent.py

```diff
@@ -78,6 +80,41 @@ def run(self) -> None:
         except Exception as e:
             self.mgr.log.error(f'Failed to run cephadm cherrypy endpoint: {e}')
 
+    def start_engine(self) -> None:
+        port_connect_attempts = 0
+        while port_connect_attempts < 1000:
+            try:
+                cherrypy.engine.start()
+                self.mgr.set_module_option('endpoint_port', self.server_port)
+                break
+            except cherrypy.process.wspbus.ChannelFailures as e:
+                if 'Address already in use' in str(e):
+                    self.mgr.log.info(
+                        f'{e}. Trying next port.')
+                    self.server_port += 1
```

Comment:
Why aren't we just using 8499, and then if a bind fails move it up to the next free port - instead of letting this be totally configurable. Ceph functions (mon, osd, mgr) on a set of known ports, so why are we not doing the same?